vaadin-cdi/src/test/java/com/vaadin/cdi/MultipleAccessIsolationTest.java

```diff
@@ -41,17 +41,19 @@ public void testConcurrentAccess() throws IOException,
         assertThat(getCount(ConcurrentUI.CONSTRUCT_COUNT), is(1));
         firstWindow.findElement(By.id(ConcurrentUI.OPEN_WINDOW)).click();
         waitForNumberOfWindowsToEqual(2);
-        Thread.sleep(500);
         List<String> handles = new ArrayList<String>(
                 firstWindow.getWindowHandles());
+        // wait for second window fully loads
+        firstWindow.switchTo().window(handles.get(1));
+        (new WebDriverWait(firstWindow, 15)).until(ExpectedConditions
+                .presenceOfElementLocated(LABEL));
+
         firstWindow.switchTo().window(handles.get(0));
```

Comment:
I see that all switchTo switches to a window already loaded.
line 51: window 0 loaded because openWindow on line 4 waits to label. Click on line 42 does not trigger any server side event listener.
line 56: window 1 loaded because of wait on line 48-49